app/src/main/java/org/mozilla/fenix/perf/PerfNavController.kt

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import androidx.annotation.VisibleForTesting
+import androidx.navigation.NavController
+import kotlinx.coroutines.Job
+import java.util.WeakHashMap
+
+/**
+ * We need to keep a weak reference to our NavController in case our activity
+ * gets re-created and we need to link our NavGraph to the NavController again.
+ * The value of this map is the job associated to the NavController which should link the NavGraph
+ * to our NavController.
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+val map = WeakHashMap<NavController, Job>()
+
+fun addNavToMap(navController: NavController, job: Job) {
```

Comment:
This code is hard to follow because implementation details are spread between two files: `HomeActivity` shouldn't handle the responsibility of adding an async nav graph job to a WeakHashMap.

Can we encapsulate this functionality in one file so it's easier to follow? For example,
```kotlin
object AsyncNavGraphProvider {
    fun startAsyncInflation(navController: NavController) { ... }
    fun blockForInflation(navController: NavController) { ... }    
}
```

As a bonus, this will make commenting what we're doing easier because right now, I think the commenting isn't clear enough.